server/src/main/java/org/apache/iotdb/db/mpp/plan/analyze/AnalyzeVisitor.java

```diff
@@ -878,7 +878,7 @@ private void analyzeDeviceToSourceTransform(Analysis analysis, QueryStatement qu
         Set<Expression> aggregationExpressions = deviceToAggregationExpressions.get(deviceName);
         Set<Expression> sourceTransformExpressions = new LinkedHashSet<>();
         for (Expression expression : aggregationExpressions) {
-          sourceTransformExpressions.addAll(expression.getExpressions());
+          sourceTransformExpressions.add(expression.getExpressions().get(0));
```

Comment:
count_if and other built-in agg functions only need the first argument, we need to consider the future, or at least add a comment about it